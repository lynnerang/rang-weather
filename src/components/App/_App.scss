.App {
  text-align: center;
  position: relative;
  box-sizing: border-box;
}

.App-logo {
  height: 20px;
  margin-right: 8px;
  margin-top: 4px;
}

.App-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 50px;
  background: linear-gradient(to bottom right, #1b4b5b, rgb(19, 54, 65));
  position: fixed;
  width: 100vw;
  z-index: 6000;

  .App-header-left {
    display: flex;
    margin-left: 15px;
  }

  h1 {
    font-family: 'Poppins', sans-serif;
    font-size: 18px;
    font-weight: 400;
    margin: 0;
  }

  .App-header-right {
    margin-right: 15px;
  }

  .fa-bars {
    font-size: 20px;
    color: white!important;

    &:hover {
      color: lightblue;
    }
  }
}

.App-nav {
  padding: 5px 0 5px 10px;
  width: 100vw;
  background: rgb(74, 126, 143);
  position: fixed;
  top: 50px;
  right: 0;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  z-index: 4;
  border-bottom: 2px solid rgb(55, 91, 104);
  box-shadow: 3px 3px 3px rgba(0, 0, 0, 0.25);

  .nav-link {
    text-decoration: none;
    margin: 20px;
    cursor: pointer;
    color: white;
    font-weight: 500;
    font-size: 20px;
  }

  hr {
    color: gray;
    width: 100%;
    opacity: .3;
  }
}

.App-main {
  padding-top: 50px;
}

.main-title {
  background: #3D6572;
  height: 26px;
  font-style: italic;
  font-size: 16px;
  font-weight: 300;
  text-align: left;
  padding-left: 10px;
  vertical-align: middle;
  padding-top: 5px;
  margin: 0;
  border-top: 3px solid rgb(24, 48, 57);
}

