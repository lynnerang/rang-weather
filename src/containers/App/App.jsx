import React, { Component } from 'react';
import { Switch, Route, NavLink } from 'react-router-dom';
import CurrentStats from '../../components/CurrentStats/CurrentStats';
import Forecast from '../Forecast/Forecast';


export class App extends Component {
  state = {
    showNav: false
  }

  render() {
    let page; 

    // !this.props.currentStats.length || !this.props.historicalData.length ?
    //   page = 'loading...'

    //   :

    const nav = this.state.showNav ? (
      <nav className="App-nav">
        <NavLink className="nav-link" to="/" onClick={() => this.setState({showNav: false})}>Home</NavLink>
        <hr/>
        <NavLink className="nav-link" to="/forecast" onClick={() => this.setState({showNav: false})}>Forecast</NavLink>
      </nav>
    ): null;
      
      page = (
        <div className="App">
          {nav}
          <header className="App-header">
            <div className="App-header-left">
              <img className="App-logo" src={require('../../images/rangweatherlogo.png')} alt="App logo" />
              <h1>RangWeather</h1>
            </div>
            <div className="App-header-right">
              <i className="fas fa-bars" onClick={() => this.setState({showNav: !this.state.showNav})}/>
            </div>
          </header>
          <main className="App-main">
            <Switch>
              <Route exact path='/' render={() => <CurrentStats />} />
              <Route exact path='/forecast' render={() => <Forecast />} />
            </Switch>
          </main>
        </div>
      );
    
    return page;
  }
}

export default App;
