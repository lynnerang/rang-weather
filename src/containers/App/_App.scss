.App {
  text-align: center;
}

.App-logo {
  height: 20px;
  margin-right: 8px;
  margin-top: 4px;
}

.App-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 12px 16px;
  opacity: .9;

  .App-header-left {
    display: flex;
  }

  h1 {
    font-family: 'Poppins', sans-serif;
    font-size: 18px;
    font-weight: 400;
    margin: 0;
  }

  .fa-bars {
    font-size: 20px;

    &:hover {
      color: lightblue;
    }
  }
}

.App-main {
  margin-top: 10px;
}

