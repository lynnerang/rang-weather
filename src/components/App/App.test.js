import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('App', () => {
  let wrapper;
  let mockEvent;

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('matches the snapshot when not showing navigation', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have the appropriate default state', () => {
    expect(wrapper.state()).toEqual({
      showNav: false
    })
  })

  it('should update the state when the nav button is clicked', () => {
    wrapper.find('.fa-bars').simulate('click');
    expect(wrapper.state()).toEqual({
      showNav: true
    })
  })

  it('matches the snapshot when showNav is set to true', () => {
    wrapper.find('.fa-bars').simulate('click');
    expect(wrapper.state()).toEqual({
      showNav: true
    })
    
    expect(wrapper).toMatchSnapshot();
  })
  
});
