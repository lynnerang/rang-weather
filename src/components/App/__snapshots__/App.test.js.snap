// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot when not showing navigation 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="App-header-left"
    >
      <img
        alt="App logo"
        className="App-logo"
        src="rangweatherlogo.png"
      />
      <h1>
        RangWeather
      </h1>
    </div>
    <div
      className="App-header-right"
    >
      <i
        className="fas fa-bars"
        onClick={[Function]}
      />
    </div>
  </header>
  <main
    className="App-main"
  >
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/forecast"
        render={[Function]}
      />
    </Switch>
  </main>
</div>
`;

exports[`App matches the snapshot when showNav is set to true 1`] = `
<div
  className="App"
>
  <nav
    className="App-nav"
  >
    <NavLink
      className="nav-link"
      onClick={[Function]}
      to="/"
    >
      Home
    </NavLink>
    <hr />
    <NavLink
      className="nav-link"
      onClick={[Function]}
      to="/forecast"
    >
      Forecast
    </NavLink>
  </nav>
  <header
    className="App-header"
  >
    <div
      className="App-header-left"
    >
      <img
        alt="App logo"
        className="App-logo"
        src="rangweatherlogo.png"
      />
      <h1>
        RangWeather
      </h1>
    </div>
    <div
      className="App-header-right"
    >
      <i
        className="fas fa-bars"
        onClick={[Function]}
      />
    </div>
  </header>
  <main
    className="App-main"
  >
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/forecast"
        render={[Function]}
      />
    </Switch>
  </main>
</div>
`;
